stories/components/tanstack/full_example.stories.tsx

```diff
@@ -0,0 +1,291 @@
+import { Classes } from '@blueprintjs/core';
+import styled from '@emotion/styled';
+import { revalidateLogic } from '@tanstack/react-form';
+import type { FormEvent } from 'react';
+import { action } from 'storybook/actions';
+import { match } from 'ts-pattern';
+import { z } from 'zod';
+
+import { useForm, useSelect, withForm } from '../../../src/components/index.js';
+
+export default {
+  title: 'Forms / Form / Tanstack',
+};
+
+const Fieldset = styled.fieldset`
+  & legend {
+    padding-right: 5px;
+    background: white;
+    font-weight: 550;
+  }
+
+  padding-top: 5px;
+  padding-bottom: 10px;
+
+  & + & {
+    border-top: none;
+    margin-top: calc(-8px + -10px);
+  }
+
+  &:has(+ &) {
+    padding-bottom: 20px;
+  }
+`;
+
+const cssColSpan = Array.from(
+  { length: 6 },
+  (_, i) => `& .col-span-${i} {
+    grid-column: span ${i} / span ${i};
+  }`,
+).join('\n\n');
+
+const GridGroup = styled.div`
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  gap: 0.5em;
+  justify-items: stretch;
+
+  ${cssColSpan}
+
+  & > .contents {
+    display: contents;
+  }
+
+  & > .${Classes.FORM_GROUP} {
+    margin: 0;
+  }
+
+  /* tweaks styling for Select inline in a grid */
+  & .${Classes.FORM_CONTENT} {
+    flex: 1;
+
+    .${Classes.POPOVER_TARGET} {
+      width: 100%;
+
+      .${Classes.BUTTON} {
+        justify-content: space-between;
+      }
+    }
+  }
+
+  /* tweaks for inputs in cell grid */
+  & .${Classes.FORM_CONTENT} .${Classes.CONTROL_GROUP} .${Classes.INPUT_GROUP} {
+    flex-shrink: 1;
+    flex-grow: 1;
+  }
+`;
```

Comment:
Ditto, this is too complicated and does not even look good

<img width="568" height="412" alt="CleanShot 2025-08-29 at 15 03 41@2x" src="https://github.com/user-attachments/assets/f411398a-93f7-4a94-897d-2bf1ae91d1b8" />